### DISCUSSION TITLE
erro ao instalar no Delphi 10.2 | error when installing in Delphi 10.2

### DISCUSSION THREAD

**Discussion by Ronierys2** [project-jedi/jcl#153] [Q&A]
![image](https://github.com/user-attachments/assets/538666eb-b7e5-47a2-b03f-88c0483b81ca)

ao tentar fazer a instalação no Delphi 10.2 Windows 10 64btis tenho a seguinte mensagem de erro acima. 

English:

When trying to install Delphi 10.2 Windows 10 64btis I have the following error message above.

**Comment by obones** [comment]
Hello,

Which version did you use? If this was with the GIT content, could you retry with the latest as of today?
There were changes made to support C++ Builder Modern Win64 compiler that were not complete until yesterday so maybe you updated in between.
If the issue is still there, please post the log file content for me to have a look.

**Reply by DanielPetatCes** [reply]
Hello,

i've got the same error with Delphi 10.1, using the latest source code.
When I revert the changes from commit ae9a575b411792c872076f95bb7603439aa2d811  everything works fine.

I tried debugging the installer with another delphi version and found the following. 
`TJclBorRADToolInstallation.CheckPlatform` is called with `APlatform = bpWin32c`, but Personalities says, I have only Delphi32 and Delphi64.
See attached screenshot.
(I just bought Delphi, not the full RAD studio with the C compiler included)
Hope this helps.

Kind regards
Daniel

![2025-02-19_14h29_06](https://github.com/user-attachments/assets/c0351002-560a-45d0-8dce-d092d50c9ee5)

**Reply by obones** [reply]
Well, this is very strange because this code is only called [here](https://github.com/project-jedi/jcl/blob/master/jcl/source/common/JclIDEUtils.pas#L1965) when `CommandLineTools` contains `clBcc32c` and this is happens with this code from `JclIDEUtils.TJclBorRADToolInstallation.Create` (See [here](https://github.com/project-jedi/jcl/blob/master/jcl/source/common/JclIDEUtils.pas#L1856)):

```pascal
  if FileExists(BinFolderName + BCC32CExeName) then
    Include(FCommandLineTools, clBcc32c);
```

Which means that there exists a file called `bcc32c.exe` in your installation folder.
I checked on my side of things, with a machine where I only have Delphi 10.4 and Delphi 11, no signs of a `bcc32c.exe` file or even a `bcc32.exe` which is expected with a Delphi only installation.

**Reply by DanielPetatCes** [reply]
Strange..  On my machine, and another one where I have 10.1 installed, there is indeed a `bcc32c.exe` file in the Delphi installation folder. But no `bcc32.exe`
Maybe it was a bug in the setup?
When I rename this file, so the JCL installer cannot find it, the installation is successful. So I have a workaround, thank you.

By the way, just to complete the confusion:
In Delphi 12.2, the only bcc*  file I can find is `bcc64x.exe`  in the `bin64`folder.  Nothing in the usual `bin` folder.

**Reply by obones** [reply]
Thanks, it has been confirmed that some update packages do not check the installed personalities and deploy all command line compilers.
The JCL code has been updated to take this into account

**Comment by rodrigoceren** [comment]
I'm having problems too! Version 10.3 32bits. The other versions install without problems, but this version the problem persists, even getting it from the daily repository and from getit.
What can I do to continue?

![image](https://github.com/user-attachments/assets/258ee5fd-7a09-4e21-b7dc-cf4a87810e09)

**Reply by obones** [reply]
Just like with others, please check if there is a file called `brcc32c.exe` in the bin folder of your Delphi installation

**Reply by rodrigoceren** [reply]
Thank you! The solution I found was to create an extra line in the Jcl installer.

**Reply by obones** [reply]
Would you care to share what this extra line is?

**Reply by rodrigoceren** [reply]
<img width="1095" alt="image" src="https://github.com/user-attachments/assets/1e7d2929-cedf-409d-95e5-7daf17847607" />
I just commented on this line

**Reply by obones** [reply]
Ah yes, I see, you simply hide the reason why the exception is triggered, but there has to be a better way as this test is important to detect incoherence in installation choices.

**Reply by ronaldhoek** [reply]
I got the same error on Delphi (RAD Studio) 10.3 32-bit (64-bit is okay). 
Installing JCL for Delphi (RAD Studio) XE2 (32/64-bit) was also no problem.

None of my Delphi versions (XE2 or 10.3) contains the file `brcc32c.exe`

**Reply by ronaldhoek** [reply]
This 'BUG' is introduced in the commit:

Revision: ae9a575b411792c872076f95bb7603439aa2d811
Author: OBones <obones@free.fr>
Date: 17-12-2024 10:25:35
Message: fix: The C++ Builder 32 bit clang base compiler requires options in suffixed values

Modified: jcl/source/common/JclCompilerUtils.pas
Modified: jcl/source/common/JclIDEUtils.pas

**Comment by ronaldhoek** [comment]
@obones I noticed the PUSH https://github.com/project-jedi/jcl/commit/dc79734383955ea6df3ad1762db64c741ba2fd87 for the current `master` and checked it just now.

No more errors when installing for Delphi (RAD Studio) 10.3 32-bit (of any of my other personalities)

**Reply by ronaldhoek** [reply]
The latest PUSH https://github.com/project-jedi/jcl/commit/ed19273c1c585514b21dcd373b547a2c942f17ac does not give any new problems too ;)

**Reply by rodrigoceren** [reply]
Thanks for all!

**Reply by obones** [reply]
Yes, these two commits are the bare minimum for the installer to work properly.
Note that the presence of `bcc32c.exe` when only the Delphi personality has been installed was an error in the patch/update system from Embarcadero.

**Reply by ronaldhoek** [reply]
Nice work @obones - keeping things backwards compatible is hard, but is very much appreciated!
